### PR TITLE
[#2789] Model "Mijn Vragen" in OpenKlant2 as klantcontacten

### PR DESCRIPTION
**Pull request by swrichards**
[Taiga 2789](https://taiga.maykinmedia.nl/project/open-inwoner/task/2789).

### CODE REVIEW

File: src/open_inwoner/openklant/services.py [hunk 1]
```diff
@@ -380,3 +447,177 @@ def update_partij_from_user(self, partij: Partij, user: User):
...
+            question = klantcontact_uuid_to_klantcontact_object[question_uuid]
+            try:
+                answer_uuid = answers_for_klantcontact_uuid[question_uuid]
+                answer = klantcontact_uuid_to_klantcontact_object[answer_uuid]
+            except KeyError:
+                answer = None
+
+            answer_obj = None
+            if answer:
+                answer_obj = OpenKlant22Answer.from_klantcontact(answer)
```
Comment (pi-sigma) [inline src/open_inwoner/openklant/services.py:616]:
```suggestion
                answer_obj = OpenKlant2Answer.from_klantcontact(answer)
```

File: src/open_inwoner/openklant/tests/test_openklant2_service.py [hunk 2]
```diff
@@ -269,3 +274,137 @@ def test_update_partij_from_user(self):
...
+@freezegun.freeze_time(QUESTION_DATE)
+class QuestionAnswerTestCase(Openklant2ServiceTestCase):
+
+    # @classmethod
+    # def get_config(cls):
+
+    #     return {
+    #         "use_live_service": True,
+    #         "vcr_record_mode": RecordMode.ALL,
+    #     }
```
Comment (pi-sigma) [inline src/open_inwoner/openklant/tests/test_openklant2_service.py:294]: Do we still need this?

File: src/open_inwoner/openklant/tests/test_openklant2_service.py [hunk 3]
```diff
@@ -269,3 +274,137 @@ def test_update_partij_from_user(self):
...
+                "indicatieActief": True,
+            }
+        )
+        self.persoon = self.openklant_client.partij.create_persoon(
+            data=CreatePartijPersoonDataFactory(
+                partijIdentificatie__contactnaam__voornaam="Alice",
+                partijIdentificatie__contactnaam__achternaam="McAlice",
+            )
+        )
+        self.another_persoon = self.openklant_client.partij.create_persoon(
```
Comment (pi-sigma) [inline src/open_inwoner/openklant/tests/test_openklant2_service.py:312]:
```suggestion
        self.another_persoon = self.openklant_client.partij.create_persoon(
```
```suggestion
        self.andere_persoon = self.openklant_client.partij.create_persoon(
```

File: src/open_inwoner/openklant/tests/test_openklant2_service.py [hunk 4]
```diff
@@ -269,3 +274,137 @@ def test_update_partij_from_user(self):
...
+                answer=None,
+                nummer=klantcontact["nummer"],
+                question_kcm_uuid=klantcontact["uuid"],
+                question="A question asked by Alice",
+                plaatsgevonden_op=QUESTION_DATE,
+            ),
+        )
+
+    def test_get_questions(self):
+        for persoon in (self.persoon, self.another_persoon):
```
Comment (pi-sigma) [inline src/open_inwoner/openklant/tests/test_openklant2_service.py:381]:
```suggestion
        for persoon in (self.persoon, self.andere_persoon):
```

**Review by pi-sigma — CHANGES_REQUESTED** [review]

File: src/open_inwoner/openklant/services.py [hunk 5]
```diff
@@ -380,3 +447,177 @@ def update_partij_from_user(self, partij: Partij, user: User):
                 f"updated Partij from user with fields: {', '.join(sorted(updated_fields))}",
                 content_object=user,
             )
+
+    def create_question(
+        self, partij: Partij, question: str, subject: str
+    ) -> OpenKlant2Question:
+
+        if len(question.rstrip()) == 0:
+            raise ValueError("You must provide a question")
```
[the remark this answers is not in the document]
Comment (pi-sigma) [reply src/open_inwoner/openklant/services.py:456]: No, the assumption was that these errors were displayed to the client at some point

**Review by pi-sigma — APPROVED** [review]

File: src/open_inwoner/openklant/services.py [hunk 6]
```diff
@@ -380,3 +447,177 @@ def update_partij_from_user(self, partij: Partij, user: User):
...
+            }
+        )
+
+        self.client.betrokkene.create(
+            data={
+                "rol": "klant",
+                "hadKlantcontact": {"uuid": answer_klantcontact["uuid"]},
+                "initiator": True,
+                "wasPartij": {"uuid": partij["uuid"]},
+                "organisatienaam": "Open Inwoner Platform",
```
Comment (alextreme) [inline src/open_inwoner/openklant/services.py:528]: graag op self zetten als constant zodat we dit later configureerbaar maken

File: src/open_inwoner/openklant/services.py [hunk 7]
```diff
@@ -380,3 +447,177 @@ def update_partij_from_user(self, partij: Partij, user: User):
...
+            }
+        )
+        logger.info("Created betrokkene: %s", betrokkene["uuid"])
+
+        taak = self.client.interne_taak.create(
+            data={
+                "aanleidinggevendKlantcontact": {"uuid": klantcontact["uuid"]},
+                "toelichting": "Beantwoorden vraag",
```
Comment (alextreme) [inline src/open_inwoner/openklant/services.py:490]: graag op self zetten als constant zodat we dit later configureerbaar maken

**Pull request merged**